highfidelity/hifi-content
# WL 21727 - Music Visualizer Updates (marketplace item)

> Current asset is here:
> http://mpassets.highfidelity.com/ced4e659-3a5a-442b-8de4-1acdf68e484e-v1/musicVisualizer.js
> Please update this script for better particle appearance (currently they're too fast, fuzzy, etc) and make sure it fully cleans up once script is closed. Resubmit to Marketplace when complete.
> [Worklist 21727](https://worklist.net/21727)

## Test Plan
1. Run the script from URL here: `http://mpassets.highfidelity.com/94c95207-e34c-48c8-afae-ad7dd9499a69-v1/musicVisualizer.js`
2. Confirm 'MUS VIS' gets added to your Toolbar/Tablet.
3. Supply a WAV file as the file and go through each effect.
4. Make sure there's no weird visual glitches and that it should look nice and smooth.
5. Click "Clean Up" in the app and confirm it deletes all existing Visualizers.
6. With visualizers in your sandbox, connect to a different domain then back to your sandbox.
7. Confirm that the Visualizers are still there, and can be cleared using the Clean Up button in the app.
8. Confirm that restarting the script also deletes the Visualizers.

![example](https://cain.sh/i/7v7S.png)
